app/controllers/promo_registrations_controller.rb

```diff
@@ -14,20 +14,11 @@ def create
     @publisher = current_publisher
     @publisher.promo_enabled_2018q1 = true
     @publisher.save!
-    @publisher_has_verified_channel = @publisher.has_verified_channel?
-
-    if @publisher_has_verified_channel
-      if @publisher.channels.where(verified: true).count > 5
-        Promo::RegisterPublisherForPromoJob.perform_later(publisher: @publisher)
-        redirect_to home_publishers_path, notice: t("promo.activated.please_wait")
-      else
-        Promo::PublisherChannelsRegistrar.new(publisher: @publisher).perform
-        @promo_enabled_channels = @publisher.channels.joins(:promo_registration)
-        PromoMailer.promo_activated_2018q1_verified(@publisher, @promo_enabled_channels).deliver
-      end
-    else
-      PromoMailer.promo_activated_2018q1_unverified(@publisher).deliver
+    current_publisher.channels.left_joins(:promo_registration).where(promo_registrations: {id: nil}).find_each do |channel|
+      channel.register_channel_for_promo # Callee does a check
     end
+    @promo_enabled_channels = @publisher.channels.joins(:promo_registration)
```

Comment:
When a user adds channels and then activates the promo this code doesn't behave as expected. 
There's a race condition where the jobs haven't completed to show the referral codes on the screen so the user gets a list of blank referral codes.

<img src="https://user-images.githubusercontent.com/5459225/67584079-7f604b80-f712-11e9-8b74-90e008d08350.png" width="300" />
Then if you refresh the screen you get this
<img src="https://user-images.githubusercontent.com/5459225/67584181-ac146300-f712-11e9-94bc-2000d050d8cc.png" width="300" />

Reply:
Will fix to do promo creation as a background job